### PR TITLE
go mongo driver wireframe

### LINKED ISSUE #4: Setting up mongo driver and interface

### PR DESCRIPTION
**Pull request by gzcharleszhang**
closes #4

### CODE REVIEW

File: internal/app/db/db.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"os"
+)
+
+func NewSession(ctx context.Context) (*Session, error) {
```
Comment (gzcharleszhang) [inline internal/app/db/db.go]: This function creates a connection to the mongo server

File: internal/app/db/sessions.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package db
+
+import "go.mongodb.org/mongo-driver/mongo"
+
+type Session struct {
+	Client *mongo.Client
+}
+
+func (sess Session) Users() *mongo.Collection {
```
Comment (gzcharleszhang) [inline internal/app/db/sessions.go]: methods that abstracts away the collection naming details

**Comment by gzcharleszhang** [comment]
@infinitecold `db.go` and `sessions.go` are the only files i added with code, the rest are just the packages, don't worry about them

File: .env [hunk 3]
```diff
@@ -0,0 +1,3 @@
+MONGO_HOST=uwtimeline-rvzny.mongodb.net/test?retryWrites=true&w=majority
+MONGO_USERNAME=Stephen_New
```
Comment (infinitecold) [inline .env]: facts

File: .env [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+MONGO_PASSWORD=vVQux86R0pzVcLtA
```
Comment (infinitecold) [inline .env]: should we be pushing this to github

Comment (gzcharleszhang) [reply .env]: Shit

Comment (infinitecold) [reply .env]: https://help.github.com/en/articles/removing-sensitive-data-from-a-repository

Comment (gzcharleszhang) [reply .env]: yeah forgot to include it in gitignore

**Pull request merged**